common/types/endpoint.go

```diff
@@ -138,6 +142,60 @@ type Endpoint struct {
 	Consumable       *Consumable           `json:"consumable"`
 	PolicyMap        *policymap.PolicyMap  `json:"-"`
 	Opts             *BoolOptions          `json:"options"` // Endpoint bpf options.
+	Status           EndpointStatus        `json:"status"`
+}
+
+type statusLog struct {
+	Status    Status    `json:"status"`
+	Timestamp time.Time `json:"timestamp"`
+}
+
+type EndpointStatus struct {
+	Log     *[maxLogs]*statusLog `json:"log"`
+	Index   uint8                `json:"index"`
```

Comment:
You limit this both with the `uint8` and maxLogs. Chaning maxLogs down the line will not work. I suggest to use int here and limit exclusively by maxLogs.